#!/bin/sh -e
##
## 04_make_minus_j_fix.dpatch by Steve Kowalik and Colm Buckley
##
## DP: Fix building with 'make -j'.
## DP: A version of this patch was submitted upstream:
## DP:   https://bugtrack.alsa-project.org/alsa-bug/view.php?id=1370
## DP: If and when it is applied, this patch can be reduced to
## DP:   s/echo $(SNDVERSIONS).tmp/mktemp $(SNDVERSIONS).XXXXXX/

. debian/patches/patch-opts

if [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi

case "$1" in
       -patch) patch $patch_opts -p1 < $0;;
       -unpatch) patch $patch_opts -p1 -R < $0;;
        *)
                echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
                exit 1;;
esac

exit 0

@DPATCH@
diff -urNad alsa-driver-1.0.8/Rules.make /tmp/dpep.jldRsB/alsa-driver-1.0.8/Rules.make
--- alsa-driver-1.0.8/Rules.make	2004-12-09 20:39:11.000000000 +0100
+++ /tmp/dpep.jldRsB/alsa-driver-1.0.8/Rules.make	2005-01-16 15:04:22.000000000 +0100
@@ -287,7 +287,7 @@
 	genksyms_smp_prefix := 
 endif
 
-$(MODINCL)/$(MODPREFIX)%.ver: %.c
+$(MODINCL)/$(MODPREFIX)%.ver: %.c update-sndvers
 	@if [ ! -r $(MODINCL)/$(MODPREFIX)$*.stamp -o $(MODINCL)/$(MODPREFIX)$*.stamp -ot $< ]; then \
 		if [ ! -f $(CONFIG_SND_KERNELDIR)/include/linux/modules/$*.stamp ]; then \
 		echo '$(CC) -D__KERNEL__ $(CFLAGS) $(EXTRA_CFLAGS) -E -D__GENKSYMS__ $<'; \
@@ -308,7 +308,8 @@
 endif # export-objs 
 
 define update-sndvers
-	@(echo "#ifndef _LINUX_SNDVERSIONS_H"; \
+	(tmpfile=`mktemp $(SNDVERSIONS).XXXXXX`; \
+	(echo "#ifndef _LINUX_SNDVERSIONS_H"; \
 	  echo "#define _LINUX_SNDVERSIONS_H"; \
 	  echo "#include <linux/modsetver.h>"; \
 	  cd $(TOPDIR)/include/modules; \
@@ -316,14 +317,14 @@
 	    if [ -f $$f ]; then echo "#include \"modules/$${f}\""; fi; \
 	  done; \
 	  echo "#endif"; \
-	) > $(SNDVERSIONS).tmp
-	@if [ -r $(SNDVERSIONS) ] && cmp -s $(SNDVERSIONS) $(SNDVERSIONS).tmp; then \
+	) > $${tmpfile}; \
+	if [ -r $(SNDVERSIONS) ] && cmp -s $(SNDVERSIONS) $${tmpfile}; then \
 		echo $(SNDVERSIONS) was not updated; \
-		rm -f $(SNDVERSIONS).tmp; \
+		rm -f $${tmpfile}; \
 	else \
 		echo $(SNDVERSIONS) was updated; \
-		mv -f $(SNDVERSIONS).tmp $(SNDVERSIONS); \
-	fi
+		mv -f $${tmpfile} $(SNDVERSIONS); \
+	fi)
 endef
 
 $(SNDVERSIONS):
